cardinal/ecs/transaction/transaction.go

```diff
@@ -18,6 +20,10 @@ type TxID struct {
 	Index      uint64
 }
 
+func (id TxID) String() string {
+	return fmt.Sprintf("%s-%d", id.PersonaTag, id.Index)
```

Comment:
What happens if the transaction doesn't involve the persona tag of the sender? i.e. register persona transaction

Reply:
PersonaTag should always be defined (https://github.com/Argus-Labs/world-engine/blob/main/cardinal/server/utils.go#L98-L100).

That being said, two different clients (say Nakama and some random smart contract) could both request the same persona tag with (luckily) the same nonce, and generate identical IDs. So maybe <personatag>-<nonce> isn't going to work.

Reply:
What is the persona tag used when I call register persona?

Reply:
A signed payload is:
```
type SignedPayload struct {
	PersonaTag string
	Namespace  string
	Nonce      uint64
	Signature  string          // hex encoded string
	Body       json.RawMessage // json string
}
```
Which includes a PersonaTag. The linked code above verified that the PersonaTag is not empty. 

Digging more into the code, it looks like we never actually verify that the PersonaTag in the signature matches the PersonaTag in the message body. 

That's wrong (i'll fix in this PR), but still, it shouldn't be empty.

Reply:
This PR doesn't include the refactoring of SignedPayload to RawTransaction/SignedTransaction: https://linear.app/arguslabs/issue/CAR-159/refactor-signsignedpayload-into-rawtransaction-and-signedtransaction